test/storage.js

```diff
@@ -0,0 +1,64 @@
+var lastTest = "",
+	done = false,
+	mockStorage = {
+		_store: Object.create( null ),
+		setItem: function( key, value ) {
+			this._store[ key ] = value;
+			this.length++;
+		},
+		getItem: function( key ) {
+			return key in this._store ? this._store[ key ] : null;
+		},
+		removeItem: function( key ) {
+			if ( key in this._store ) {
+				delete this._store[ key ];
+				this.length--;
+			}
+		},
+		key: function( i ) {
+			return Object.keys( this._store )[ i ];
+		},
+		length: 0
+	};
+
+mockStorage.setItem( "qunit-test-Storage-Should be run first", 1 );
+QUnit.config.storage = mockStorage;
+
+QUnit.module( "Storage", function() {
+	QUnit.test( "Passing test", function( assert ) {
+		assert.strictEqual( lastTest, "Should be run first" );
+		mockStorage.setItem( "qunit-test-Storage-Passing test", true );
+		assert.ok( true );
+	} );
+
+	QUnit.test( "Removes passing tests from storage", function( assert ) {
+		assert.strictEqual( mockStorage.getItem( "qunit-test-Storage-Passing test" ), null );
+	} );
+
+	// Verifies test reordering by storage values
+	QUnit.test( "Should be run first", function( assert ) {
+		assert.strictEqual( lastTest, "" );
+		lastTest = "Should be run first";
+	} );
+} );
+
+QUnit.done( function() {
+	if ( done ) {
+		return;
+	}
+
+	done = true;
+
+	mockStorage.setItem( "qunit-test-should-remove", true );
+	mockStorage.setItem( "should-not-remove", true );
+	mockStorage.setItem( "qunit-test-should-also-remove", true );
+
+	// TODO: Is there a way to test this that will show up in CI?
```

Comment:
I'm surprised it works on Node; this looks like a problem there as well: https://github.com/jquery/qunit/blob/master/build/tasks/test-on-node.js#L80 . Tools just aren't built for runs to end more than once: https://github.com/browserstack/browserstack-runner/blob/f5689b8d03bc4695e2f8a536d94b3f7316bd78ca/lib/_patch/reporter.js#L20 .

Maybe we can instead invoke the file twice, the second time with a query parameter/environment variable/etc. that we look for to alter behavior?